mkdocs/docs/metrics/defining-iter8-metrics.md

```diff
@@ -0,0 +1,640 @@
+          urlTemplate: http://myprometheusservice.com/api/v1
+        ```
+
+    === "Basic auth"
+        Suppose Prometheus, in conjunction with the reverse proxy, is set up to enforce basic auth with the following credentials:
+
+        ```yaml
+        username: produser
+        password: t0p-secret
+        ```
+
+        You can then enable Iter8 to query this Prometheus instance as follows.
+
+        1. **Create secret:** Create a Kubernetes secret that contains the authentication information. In particular, this secret needs to have the `username` and `password` fields in the `data` section with correct values.
+        ```shell
+        kubectl create secret generic promcredentials -n myns --from-literal=username=produser --from-literal=password=t0p-secret
+        ```
+
+        2. **Create RBAC rule:** Provide the required permissions for Iter8 to read this secret. The service account `iter8-analytics` in the `iter8-system` namespace will have permissions to read secrets in the `myns` namespace.
+        ```shell
+        kubectl create rolebinding iter8-cred --clusterrole=iter8-secret-reader-analytics --serviceaccount=iter8-system:iter8-analytics --namespace=myns
+        ```
+
+        3. **Define metric:** When defining the metric, ensure that the `authType` field is set to `Basic` and the appropriate `secret` is referenced.
+
+        ```yaml linenums="1"
+        apiVersion: iter8.tools/v2alpha2
+        kind: Metric
+        metadata:
+          name: request-count
+        spec:
+          description: A Prometheus example
+          provider: prometheus
+          params:
+          - name: query
+            value: >-
+              sum(increase(revision_app_request_latencies_count{service_name='${name}',${userfilter}}[${elapsedTime}s])) or on() vector(0)
+          type: Counter
+          authType: Basic
+          secret: myns/promcredentials
+          jqExpression: ".data.result[0].value[1] | tonumber"
+          urlTemplate: https://my.secure.prometheus.service.com/api/v1
+        ```
+
+    ??? hint "Brief explanation of the `request-count` metric"
+        1. Prometheus enables metric queries using HTTP GET requests. `GET` is the default value for the `method` field of an Iter8 metric. This field is optional; it is omitted in the  definition of `request-count`, and defaulted to `GET`.
+        2. Iter8 will query Prometheus during each iteration of the experiment. In each iteration, Iter8 will use `n` HTTP queries to fetch metric values for each version, where `n` is the number of versions in the experiment[^2].
+        3. The HTTP query used by Iter8 contains a single query parameter named `query` as [required by Prometheus](https://prometheus.io/docs/prometheus/latest/querying/api/). The value of this parameter is derived by [substituting the placeholders](#placeholder-substitution) in the value string.
+        4. The `urlTemplate` field provides the URL of the prometheus service.
```

Comment:
I suggest also explaining the `jqExpression` field here as well.

Reply:
Done.